src/sql/workbench/contrib/query/browser/queryEditor.ts

```diff
@@ -272,9 +272,11 @@ export class QueryEditor extends EditorPane {
 			this.connectionManagementService.getProviderIdFromUri(this.input?.uri) ||
 			this.connectionManagementService.getDefaultProviderId();
 		// TODO: Make it more generic, some way for extensions to register the commands it supports
-		if ((providerId === 'KUSTO') || this.modeService.getExtensions('Kusto').indexOf(fileExtension) > -1) {
+		if ((providerId === 'KUSTO') || this.modeService.getExtensions('Kusto').indexOf(fileExtension) > -1 ||
+			providerId === 'LOGANALYTICS' || this.modeService.getExtensions('LogAnalytics').indexOf(fileExtension) > -1) {
 			if (this.input instanceof UntitledQueryEditorInput) {		// Sets proper language mode for untitled query editor based on the connection selected by user.
-				this.input.setMode('kusto');
+				let mode = providerId === 'KUSTO' ? 'kusto' : 'loganalytics';
```

Comment:
You should be checking the file extension as well

Reply:
Pinging this again - the initial check for this block checks both the provider ID or the language but this only checks the provider. And because of how you made this logic I believe that if you open a kql file then it's going to open with mode set to loganalytics. You should verify whether this happens (but in general I would still suggest two separate checks before the if to determine whether it's a kusto or log analytics that you can then use both in the if and here)